### PR TITLE
Simplify `impAssignStruct[Ptr]` and `gtNewBlkOpNode`

### PR DESCRIPTION
**Pull request by SingleAccretion**
1) Remove redundant / dead code from `impAssignStruct[Ptr]`
2) Simplify various aspects of `gtNewBlkOpNode`.

The majority of the [diffs](https://dev.azure.com/dnceng-public/public/_build/results?buildId=81443&view=ms.vss-build-web.run-extensions-tab) come from the usage of `gtNewStructVal`: it simplifies some more `OBJ(ADDR(LCL_VAR))` forms early (leading, in one case, to more inlining), and fixes the bug where we did not set `GTF_GLOB_REF` and `GTF_EXCEPT` on the LHS indir (this is the source of a small number of regressions).

### CODE REVIEW

**Comment by SingleAccretion** [comment]
@dotnet/jit-contrib

File: src/coreclr/jit/gentree.cpp [hunk 1]
```diff
@@ -7167,7 +7171,7 @@ GenTree* Compiler::gtNewZeroConNode(var_types type)
 {
     GenTree* zero;
 
-    switch (type)
+    switch (genActualType(type))
```
Comment (BruceForstall) [inline src/coreclr/jit/gentree.cpp:7174]: A lot of callers already call `gtNewZeroConNode(genActualType(type))`. As a matter of consistency, shouldn't we generally assume "actual" types everywhere, and callers who are dealing with full types call genActualType as necessary?

**Review by BruceForstall — APPROVED** [review]
LGTM. Thanks again for your great work, and sorry for the delay.

File: src/coreclr/jit/gentree.cpp (hunk 1 above)
Comment (SingleAccretion) [reply src/coreclr/jit/gentree.cpp:7174]: I think it is an overall usability improvement for the method to accept small types. It's a bit of a "gotcha" that it didn't work for them in my opinion, and I personally spent a few write-build-SPMI cycles on fixing forgotten `genActualType`s around it (hence the change).

> A lot of callers already call `gtNewZeroConNode(genActualType(type))`

Hmm, I thought I deleted the obvious double `genActualType`s, but if some were missed, that will be fixed.

**Pull request merged**